SteamBot/Bot.cs

```diff
@@ -17,45 +17,103 @@
 
 namespace SteamBot
 {
-    public class Bot
+    public class Bot : IDisposable
     {
-        public string BotControlClass;
-        // If the bot is logged in fully or not.  This is only set
-        // when it is.
-        public bool IsLoggedIn { get; private set; }
-
-        // The bot's display name.  Changing this does not mean that
-        // the bot's name will change.
-        public string DisplayName { get; private set; }
-        private string SchemaLang { get; set; }
+        #region Bot delegates
+        public delegate UserHandler UserHandlerCreator(Bot bot, SteamID id);
+        #endregion
 
-        // The response to all chat messages sent to it.
-        public string ChatResponse;
+        #region Private readonly variables
+        private readonly SteamUser.LogOnDetails logOnDetails;
+        private readonly string schemaLang;
+        private readonly string logFile;
+        private readonly Dictionary<SteamID, UserHandler> userHandlers;
```

Comment:
The old code was

    Dictionary<ulong, UserHandler> userHandlers = new Dictionary<ulong, UserHandler>();

`userHandlers` is now always null.